src/main/java/local/alfonso/orders/service/AgentsServiceImpl.java

```diff
@@ -0,0 +1,125 @@
+import local.alfonso.orders.repos.AgentsRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.persistence.EntityNotFoundException;
+import java.util.ArrayList;
+
+@Service(value = "agentsService")
+public class AgentsServiceImpl implements AgentsService
+{
+    @Autowired
+    private AgentsRepository arepos;
+
+    @Override
+    public ArrayList<Agents> findAll() {
+        ArrayList<Agents> list = new ArrayList<>();
+        arepos.findAll().iterator().forEachRemaining(list::add);
+        return list;
+    }
+
+    @Override
+    public Agents findAgentById(long id) {
+        return arepos.findById(id).orElseThrow(() -> new EntityNotFoundException(Long.toString(id)));
+    }
+
+    @Override
+    public Agents findAgentByName(String name) {
+        Agents agent = arepos.findByAgentname(name);
+
+        if (agent == null)
+        {
+            throw new EntityNotFoundException("Agent by the name of " + name + " not found.");
+        }
+        return agent;
+    }
+
+    @Override
+    public void delete(long id) {
+        if (arepos.findById(id).isPresent())
+        {
+            arepos.deleteById(id);
+        } else
+        {
+            throw new EntityNotFoundException(Long.toString(id));
+        }
+
+    }
+
+    @Transactional
+    @Override
+    public Agents save(Agents agent) {
+        Agents newAgent = new Agents();
+
+        newAgent.setAgentname(agent.getAgentname());
+        newAgent.setWorkingarea(agent.getWorkingarea());
+        newAgent.setCommission(agent.getCommission());
+        newAgent.setPhone(agent.getPhone());
+        newAgent.setCountry(agent.getCountry());
+
+        for (Customers c : agent.getCustomers())
+        {
+            newAgent.getCustomers().add(new Customers(c.getCustname(),c.getCustcity(),c.getWorkingarea(),c.getCustcountry(),c.getGrade(),c.getOpeningamt(),c.getReceiveamt(),c.getPaymentamt(),c.getOutstandingamt(),c.getPhone(), c.getAgent()));
+        }
+
+        for(Orders o : agent.getOrders())
+        {
+            newAgent.getOrders().add(new Orders(o.getOrdamount(),o.getAdvanceamount(),o.getCustomer(),o.getAgent(),o.getOrdersescription()));
+        }
+        return arepos.save(newAgent);
+    }
+
+    @Override
+    public Agents update(Agents agent, long id) {
+        Agents currentAgent = arepos.findById(id).orElseThrow(() -> new EntityNotFoundException(Long.toString(id)));
+
+        if (agent.getAgentname() != null)
+        {
+            currentAgent.setAgentname(agent.getAgentname());
+        }
+
+        if(agent.getWorkingarea() != null)
+        {
+            currentAgent.setWorkingarea(agent.getWorkingarea());
+        }
+
+        if(agent.getCommission() != currentAgent.getCommission())
+        {
+            currentAgent.setCommission(agent.getCommission());
+        }
```

Comment:
As your code is right now, this will not work and throw a `NullPointerException` if the update object's `double` type values are not supplied.

You can fix this by using the wrapper class `Double` instead (uppercase 'D') on your model's variable declarations and the corresponding getters and setters. This is important because wrapper classes like `Double` can contain `null` values, whereas primitive types like `double` cannot.